activerecord/lib/active_record/persistence.rb

```diff
@@ -380,6 +380,11 @@ def destroy!
     # Any change to the attributes on either instance will affect both instances.
     # If you want to change the sti column as well, use #becomes! instead.
     def becomes(klass)
+      unless self.class <= klass || klass < self.class || self.class.table_name == klass.table_name
+        ActiveSupport::Deprecation.warn \
+          "Passing irrelevant class to `becomes' is now deprecated and will result in ArgumentError in Rails 6.1. " \
```

Comment:
I might skip most of this sentence and put the second sentence first since it highlights the important part which is the parent-child relationship.

```ruby
"Please make sure that #{self.class.name} and #{klass.name} are in inheritance relationship or share same table via `table_name'."
"Passing a class to `becomes` without an inheritance relationship is deprecated and will result in an ArgumentError in Rails 6.1."
```
Thoughts?

Reply:
> Passing a class to `becomes` without an inheritance relationship is deprecated

This is not accurate.
For example, [developer.rb](https://github.com/rails/rails/blob/master/activerecord/test/models/developer.rb) has classes that is not in STI relationship but share shame table.
Deprecating such usage broke the test.
That's why I add `self.class.table_name == klass.table_name` and this phrasing.

Swapping sentence make sense for me.